src/fieldViews/__tests__/AttributeFieldView.spec.ts

```diff
@@ -0,0 +1,77 @@
+import type { Node, NodeSpec } from "prosemirror-model";
+import { Schema } from "prosemirror-model";
+import { schema } from "prosemirror-schema-basic";
+import { createEditorWithElements } from "../../__tests__/helpers";
+import { getNodeSpecForProp } from "../../nodeSpec";
+import { AttributeFieldView } from "../AttributeFieldView";
+
+const createInnerView = jest.fn();
+const updateInnerView = jest.fn();
+
+type TestAttributeFields = { value: boolean };
+
+class TestAttributeFieldView extends AttributeFieldView<TestAttributeFields> {
+  public static propName = "checkbox" as const;
+  public static defaultValue = "default";
+
+  public getNodeValue(node: Node): TestAttributeFields {
+    return node.attrs.fields as TestAttributeFields;
+  }
+
+  protected createInnerView(fields: TestAttributeFields) {
+    createInnerView(fields);
+  }
+
+  protected updateInnerView(fields: TestAttributeFields) {
+    updateInnerView(fields);
+  }
+}
+
+const testSchema = new Schema({
+  nodes: {
+    doc: schema.nodes.doc,
+    text: schema.nodes.text,
+    ...(getNodeSpecForProp("doc", "testField", {
+      type: "checkbox",
+      defaultValue: { value: false },
+    }) as { testField: NodeSpec }),
+  },
+});
+
+describe("AttributeFieldView", () => {
+  beforeEach(() => {
+    createInnerView.mockReset();
+    updateInnerView.mockReset();
+  });
+
+  it("should pass the correct value to its inheritors on createInnerView", () => {
+    const { view } = createEditorWithElements([]);
+    const node = testSchema.nodes.testField.create({
+      type: "checkbox",
+      fields: { value: true },
+    });
+    new TestAttributeFieldView(node, view, () => 0, 0, { value: false });
+
+    expect(createInnerView.mock.calls[0]).toEqual([{ value: true }]);
+  });
+
+  it("should pass the correct value to its inheritors on updateInnerView", () => {
+    const { view } = createEditorWithElements([]);
+    const node = testSchema.nodes.testField.create({
+      type: "checkbox",
+      fields: { value: false },
+    });
+
+    const fieldView = new TestAttributeFieldView(node, view, () => 0, 0, {
+      value: false,
+    });
+
+    const newNode = testSchema.nodes.testField.create({
+      fields: { value: true },
+    });
+
+    fieldView.update(newNode, 0);
+
+    expect(updateInnerView.mock.calls[0]).toEqual([{ value: true }]);
```

Comment:
Could we also use `getNodeValue` to confirm the update has worked?

Reply:
`getNodeValue` operates on a passed node, it doesn't reveal the value of the instance's current node – so I don't think this will help. It should probably be static. It's a bit of an appendix right now, and unless we refactor it soon it might be worth removing to avoid confusion!

Reply:
Could we find the updated node and check its value?

Reply:
It's a private member of this class – and we already have access to the node as we're passing it in, so we'd just be checking our own homework, if that makes sense. We could make the node property public, via a getter, perhaps? This'd be the only thing that used it for now, mind.

It's pretty interesting – if we did check this, we'd discover that our `node` value is indeed never updated! But this doesn't really matter for this class or its children – as only the values of the node are relevant. This is probably unexpected behaviour were we to rely on the node value elsewhere, mind. I'm tempted to rename `node` `initialNode` instead of adding a test, to make this clearer?

Reply:
(The only reason we need to keep a node reference around at all is to match the incoming node to the initial node's `type`. With some creative thinking we could probably remove even this dependency.)

Reply:
Sorry I'd slightly got the wrong end of the stick, I had meant we could check the ProseMirror version of that node's value but I was thinking of changes coming out of the field view. 

What you're suggesting make sense, from what I can see we're only concerned with the NodeType so we could store that instead.

Reply:
Gotcha. Yeah, reckon the latter makes sense and is least surprising, I'll do that – really useful feedback.